### ISSUE TITLE
Unable to use with Servant Server 0.18.1

### ISSUE THREAD

**Issue by oco-adam** [myfreeweb/magicbane#27]
Hi, I'm new to Haskell and Magicbane seems like a really nice collection of things, especially to simplify adding some essentials to Servant - so thanks very much for the library!

However, I'm trying to use Magicbane with the latest Servant Server (0.18.1) and I'm getting the following compilation error:

```
servant-server                    > Registering library for servant-server-0.18.1..
magicbane                         > configure
magicbane                         > Configuring magicbane-0.4.1...
magicbane                         > build
magicbane                         > Preprocessing library for magicbane-0.4.1..
magicbane                         > Building library for magicbane-0.4.1..
magicbane                         > [ 1 of 10] Compiling Magicbane.App
magicbane                         > 
magicbane                         > /tmp/stack-c4a82860c00c798d/magicbane-0.4.1/library/Magicbane/App.hs:22:37: error:
magicbane                         >     • Could not deduce (HasContextEntry
magicbane                         >                           (ψ .++ DefaultErrorFormatters) ErrorFormatters)
magicbane                         >         arising from a use of ‘serveWithContext’
magicbane                         >       from the context: HasServer χ ψ
magicbane                         >         bound by the type signature for:
magicbane                         >                    magicbaneApp :: forall β χ (ψ :: [*]).
magicbane                         >                                    HasServer χ ψ =>
magicbane                         >                                    Proxy χ -> Context ψ -> β -> ServerT χ (RIO β) -> Application
magicbane                         >         at library/Magicbane/App.hs:21:1-104
magicbane                         >     • In the expression: serveWithContext api sctx
magicbane                         >       In the expression: serveWithContext api sctx $ srv ctx
magicbane                         >       In an equation for ‘magicbaneApp’:
magicbane                         >           magicbaneApp api sctx ctx actions
magicbane                         >             = serveWithContext api sctx $ srv ctx
magicbane                         >             where
magicbane                         >                 srv c
magicbane                         >                   = hoistServerWithContext
magicbane                         >                       api (Proxy @ψ) (runMagicbaneHandler c) actions
magicbane                         >    |
magicbane                         > 22 | magicbaneApp api sctx ctx actions = serveWithContext api sctx $ srv ctx
magicbane                         >    |                                     ^^^^^^^^^^^^^^^^^^^^^^^^^
magicbane                         > 
```

My reason for using a more recent version of Servant is so that I can implement custom parse errors for request inputs, as in https://docs.servant.dev/en/stable/cookbook/custom-errors/CustomErrors.html.

I realize I can probably just implement a custom FromJSON, but I was hoping for a neater solution, using Servant ErrorFormatters with Magicbane.

Can someone please suggest a workaround for this?

**Comment by valpackett** [comment]
Sorry, I haven't been actively using Haskell in quite a while, so this package is not really maintained. Pull requests welcome. (Or even repo collaborator + hackage uploader requests, if anyone is willing to maintain this.)

This message looks like you should just add the constraint it wants to the `HasServer χ ψ =>` part in that file.

**Comment by oco-adam** [comment]
OK, no worries, and thanks very much for the comment. I'll have at making a pull request to fix it. 

I noticed that this library doesn't have any tests - how do I know if I broke anything? Or in this case, can I rely on Haskell's type system being strong and expressive enough, such that if it compiles then I can assume everything is OK?

**Comment by valpackett** [comment]
You pretty much can. There's very very little logic in this library, it's not very "testable". Just check that the examples run :)

**Comment by oco-adam** [comment]
Fixed in pull request #28
